app/controllers/batch_controller.rb

```diff
@@ -0,0 +1,64 @@
+require 'json'
+
+get "/batch/upload" do
+  erb :'batches/upload'
+end
+
+post "/batch/upload" do
+  #   # response = {manipulations: parse_bill, user_data: user_data}.to_json
+  bill_data = parse_bill.to_json
+  session[:batch_id] = Batch.create(bill_data: bill_data).id
+
+ 	redirect '/batch/new'
+
+end
+
+
+
+get '/batch/new' do
+=begin
+expect ajax request to check if there is an open batch (which means bill has been uploaded)
+if so: return true. this triggers front end ajax load command.
+if not: return false. this triggers front end custom manipulation adder thing that im building right now
+
+todo: close any existing batch when admin panel is opened by deleting session id.
+=end
+
+	erb :'batches/edit_batch'
+end
```

Comment:
Why is `get '/batch/new'` rendering `erb :'batches/edit_batch'`? Try and make your templates match the route as much as you can.